cli/cmd/install.go

```diff
@@ -553,6 +554,19 @@ func (options *installOptions) allStageFlagSet() *pflag.FlagSet {
 		"Restrict the Linkerd Dashboard's default privileges to disallow Tap and Check",
 	)
 
+	return flags
+}
+
+// allStageFlagSetButNotRecordable returns flags usable for single and multi-stage  installs and
+// upgrades but which are not recordable. For multi-stage installs, users must set these flags consistently
+// across commands.
+func (options *installOptions) allStageFlagSetButNotRecordable() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("all-stage", pflag.ExitOnError)
+
+	// addon-config is present here, instead of allStageFlagSet to prevent this from being recorded
+	// and thus for upgrades the filename will be returned, and if the file is present locally
+	// configuration will be applied automatically, without user passing the flag explicitly
+	// which is not desired behaviour.
```

Comment:
Which filename? Also, there's a bit of redundancy on what the function doc already explains.

Reply:
the name of the that was passed as `--addon-config`, but update the docs to better reflect the same.